flutter/lib/src/sentry_asset_bundle.dart

```diff
@@ -55,154 +53,186 @@ class SentryAssetBundle implements AssetBundle {
-        );
-        span?.status = SpanStatus.ok();
-        completer.complete(data);
-      }, (exception, stackTrace) {
-        completer.completeError(exception, stackTrace);
-      });
+  @override
+  void clear() => _bundle.clear();
 
-      T data;
-      try {
-        data = await completer.future;
-        _setDataLength(data, span);
-        span?.status = const SpanStatus.ok();
-      } catch (e) {
-        span?.throwable = e;
-        span?.status = const SpanStatus.internalError();
-        rethrow;
-      } finally {
-        await span?.finish();
-      }
-      return data;
-    });
-  }
+  // Wrappers
 
-  Future<T> _loadStructuredBinaryDataWithTracing<T>(
-      String key, _ByteParser<T> parser) {
-    return Future<T>(() async {
-      final span = _hub.getSpan()?.startChild(
-            'file.read',
-            description:
-                'AssetBundle.loadStructuredBinaryData<$T>: ${_fileName(key)}',
-          );
-      span?.setData('file.path', key);
-      // ignore: invalid_use_of_internal_member
-      span?.origin = SentryTraceOrigins.autoFileAssetBundle;
+  Future<T> _wrapLoad<T>(
+    ISentrySpan? outerSpan,
+    String traceName,
+    String key,
+    Future<T> Function() loadFunction, {
+    void Function(ISentrySpan?)? updateInnerSpan,
+  }) {
+    final String description;
+    if (traceName == 'loadStructuredData' ||
+        traceName == 'loadStructuredBinaryData') {
+      description = 'AssetBundle.$traceName<$T>: ${_fileName(key)}';
+    } else {
+      description = 'AssetBundle.$traceName: ${_fileName(key)}';
+    }
 
-      final completer = Completer<T>();
+    final span = outerSpan?.startChild(
+      'file.read',
+      description: description,
+    );
+    span?.setData('file.path', key);
+    // ignore: invalid_use_of_internal_member
+    span?.origin = SentryTraceOrigins.autoFileAssetBundle;
 
-      // This future is intentionally not awaited. Otherwise we deadlock with
-      // the completer.
-      // ignore: unawaited_futures
-      runZonedGuarded(() async {
-        final data = await _loadStructuredBinaryDataWrapper(
-          key,
-          (value) async => await _wrapBinaryParsing(parser, value, key, span),
-        );
-        span?.status = SpanStatus.ok();
-        completer.complete(data);
-      }, (exception, stackTrace) {
-        completer.completeError(exception, stackTrace);
-      });
+    if (updateInnerSpan != null) {
+      updateInnerSpan(span);
+    }
 
-      T data;
-      try {
-        data = await completer.future;
+    return _wrapWithCompleter(
+      action: loadFunction,
+      onSuccess: (data) {
         _setDataLength(data, span);
         span?.status = const SpanStatus.ok();
-      } catch (e) {
-        span?.throwable = e;
-        span?.status = const SpanStatus.internalError();
-        rethrow;
-      } finally {
-        await span?.finish();
-      }
-      return data;
-    });
+        span?.finish(); // Do NOT await
```

Comment:
```suggestion
       unawaited(span?.finish());
```

Reply:
we can use `unawaited` instead, let's change it for all other occurrences as well, pls check that in the pr